src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.AnyMobile.cs

```diff
@@ -45,7 +45,10 @@ private HttpMessageHandler Handler
 
                         // MetricsHandler should be descendant of DiagnosticsHandler in the handler chain to make sure the 'http.request.duration'
                         // metric is recorded before stopping the request Activity. This is needed to make sure that our telemetry supports Exemplars.
-                        handler = new MetricsHandler(handler, _nativeMeterFactory, out _);
+                        if (MetricsHandler.IsGloballyEnabled())
```

Comment:
I'm not familiar with IL trimming, but if we had these feature swtiches defined in a separate class, eg. `GlobalHttpSettings`, wouldn't that be sufficient to trim away the handler entirely, so we don't need to do further checks in the handler code anymore?

Reply:
In fact linker will eliminate the whole class. But I think at least for constructor, this is how it's done.